Doc/library/functions.rst

```diff
@@ -1593,8 +1593,8 @@ are always available.  They are listed here in alphabetical order.
 
 .. function:: super([type[, object-or-type]])
 
-   Return a proxy object that delegates method calls to a parent or sibling
-   class of *type*.  This is useful for accessing inherited methods that have
+   Return a proxy object that delegates attribute access to a parent or sibling
```

Comment:
This one needs to be restored as well.

Reply:
The 1st sentence is the general definition and the 2nd sentence is the particular usage. So if we restrict the general definition to methods but widen the particular usage to attributes, isn't it weird? What about doing the opposite: general definition about attributes, particular application about methods? Since you want to stress that `super()` is most often _used_ for methods, it seems to be the most sensible. I have made the edit, please tell me what you think.

Reply:
@rhettinger  The 1st sentence is the general definition and the 2nd sentence is the particular usage. So if we restrict the general definition to methods but widen the particular usage to attributes, isn't it weird? What about doing the opposite: the general definition about attributes, the particular usage methods? Since you want to stress that `super()` is most often _used_ for methods, it seems to be the most sensible. I have made an edit (removing the `(or data attributes)` mention in the particular usage sentence), tell me what you think.

Reply:
I think this should remain.  The general facility provided by ``super`` is centered on attribute resolution; the fact that it's most commonly used for methods is more about the usage pattern.

Reply:
The only known example of this is exotic, weird, and arguably an anti-pattern:

```
>>> class A:
...     x = True
... 
>>> class B(A):
...     x = False
... 
>>> B().x
False
>>> super(B, B()).x
True
```

I'm now thinking that the core description should be left as-is and that we could add a sentence at the bottom mentioning that attribute access is also possible.

Reply:
@freddrake What do you think of @rhettinger suggestion? I am not very satisfied with it, but it is always better than not even mentioning that attribute access is possible. And @rhettinger is the expert so he should have the last word.

Reply:
The example shown is clearly not realistic, though it does work that way.

It's far more interesting when implementing an attribute using a descriptor that delegates a portion of the work to an implementation provided by a base class.  I've found this useful on a number of occasions, though I don't think any of that code is public.

While the motivation for ``super`` was definitely it's use in extending methods, the mechanism used to implement it really has nothing to do with the type of the attribute value.

Reply:
Presumably there is a reason that MRO stands for method resolution order even though it can be used for attributes as well :-)

I don't think this patch should go forward as is.  Readers are best served by emphasizing the typical and intended use cases.  Python's *super()* is weird and atypical compared to other languages.  People read this doc section to learn how to use it. They are critically dependent on these docs. Accordingly, any change for a rare use case should be in a single, separate sentence in the last paragraph.

These docs have a long history of being confusing and their present state is the result of many iterations.  It is easy to make them worse.

Reply:
Alright, that makes sense.